gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/RandomEdgeSampling.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.sampling;
+
+import org.apache.flink.api.java.DataSet;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.model.impl.functions.epgm.SourceId;
+import org.gradoop.flink.model.impl.functions.epgm.TargetId;
+import org.gradoop.flink.model.impl.functions.utils.LeftSide;
+import org.gradoop.flink.model.impl.operators.sampling.functions.EdgeRandomFilter;
+
+/**
+ * Computes an edge sampling of the graph. Retains randomly chosen edges of a given relative amount
+ * and their associated source- and target-vertices. No unconnected vertices will retain in the
+ * sampled graph.
+ */
+public class RandomEdgeSampling implements UnaryGraphToGraphOperator {
+  /**
+   * relative amount of edges in the result graph
+   */
+  private final float sampleSize;
+
+  /**
+   * seed for the random number generator
+   * if seed is null, the random generator is created without seed
+   */
+  private final long randomSeed;
+
+  /**
+   * Creates new RandomEdgeSampling instance.
+   *
+   * @param sampleSize relative preprocess size
+   */
+  public RandomEdgeSampling(float sampleSize) {
+    this(sampleSize, 0L);
+  }
+
+  /**
+   * Creates new RandomEdgeSampling instance.
+   *
+   * @param sampleSize relative sample size
+   * @param randomSeed random seed value (can be {@code null})
```

Comment:
again, can't be null. to do so you would need 'Long' instead of 'long'. But the later code snippets indicate that your are refering to '0' instead of 'null' so a fix of the java doc should be sufficient